base_layer/core/src/validation/block_validators.rs

```diff
@@ -63,23 +68,41 @@ impl Validation<Block> for StatelessBlockValidator {
     /// 1. Is there precisely one Coinbase output and is it correctly defined with the correct amount?
     /// 1. Is the accounting correct?
     fn validate(&self, block: &Block) -> Result<(), ValidationError> {
-        let block_id = format!("block #{} ({})", block.header.height, block.hash().to_hex());
-        check_block_weight(block, &self.rules.consensus_constants(block.header.height))?;
+        if block.header.height == 0 {
+            warn!(target: LOG_TARGET, "Attempt to validate genesis block");
```

Comment:
Why have we added this?
I see no reason why cant check the genesis block. It should not fail?

Reply:
1. We don't validate the genesis block anywhere in the code 
1. The genesis block is part of consensus and isn't a normal block (it doesn't necessarily follow the same rules as blocks do)
1. Many checks have to have some exception for the genesis block
e.g.

```rust
if block.height == 0 { return EverythingIsFine } else { do the real check }
```

There is a inherent security risk if a peer managed to pass a block of height 0 with whatever they wanted in, would that be added to the blockchain? If we just say that we cannot validate a genesis block, then that all goes away